### PR TITLE
Vending Machine updates

### PR DESCRIPTION
**Pull request by rabTAI**

### CODE REVIEW

File: apps/VendingMachine/DevFiles/RandomNumberGenerator.test [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { describe, it, beforeAll } from 'bun:test'
+import { web3, stringToHex } from '@alephium/web3'
+import { deployToDevnet } from '@alephium/cli'
+import { RandomNumberGenerator } from '../artifacts/ts'
+
+describe('RandomNumberGenerator', async () => {
```
Comment (not-reed) [inline apps/VendingMachine/DevFiles/RandomNumberGenerator.test:6]: can delete this file i think?

File: apps/VendingMachine/contracts/VendingMachine.ral [hunk 2]
```diff
@@ -1,45 +1,41 @@
...
-    mut mintIsPaused: Bool // true
+    mut mintIsPaused: Bool, // true
+    mut mintedFoods: [U256;20]
 ) implements INFTCollectionWithRoyalty {
 
-    event NftMinted(minter: Address, id: ByteVec, index: U256)
+    event NftMinted(minter: Address, contractIds: ByteVec, mintAmount: U256,startingIndex: U256)
 
-    // const CollectionUri = b`Vending Machine`
+    const CollectionUri = b`Vending Machine`
```
Comment (not-reed) [inline apps/VendingMachine/contracts/VendingMachine.ral:15]: this should be a url to a collection json file i believe

File: apps/VendingMachine/tsconfig.json [hunk 3]
```diff
@@ -21,5 +21,5 @@
     "types": ["bun-types"]
   },
   "exclude": ["node_modules"],
-  "include": ["src/**/*.ts", "scripts/**/*.ts", "alephium.config.ts", "artifacts/**/*.ts"]
+  "include": ["src/**/*.ts", "scripts/**/*.ts", "alephium.config.ts", "artifacts/**/*.ts", "scripts/0_deploy_Foods.ts", "scripts/1_deploy_VendingMachine.ts", "DevFiles/0_deploy_RandomNumberGenerator.ts"]
```
Comment (not-reed) [inline apps/VendingMachine/tsconfig.json:24]: it should work without adding these i think :thinking: maybe restart vscode ;)

File: apps/VendingMachine/test/VendingMachine.test.ts [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+      let newNftContractAddress = contractAddresses[i].address
+      const newNftContractState = Foods.at(newNftContractAddress)
+      console.log('NFT Index: ', (await newNftContractState.methods.getNFTIndex()).returns)
+    }
+  })
+
+  it('should  check user NFT balance ', async () => {
+    console.log('Signer 1 Address ', signer1.address)
+    console.log('Signer 1 ALPH balance: ', Number(await getAlphBalance(signer1.address, signer)) / NO_DECIMALS)
+    console.log('User NFT balance', await getTokenBalance(signer1.address, foodsContractAddress, signer))
```
Comment (not-reed) [inline apps/VendingMachine/test/VendingMachine.test.ts:172]: does this work properly? i think you are passing the token `address` here, but line 224 you are comparing the address to the `id` 
```ts
const tokenBalance = allBalances.tokenBalances?.find((t) => t.id === token)?.amount
```

File: apps/VendingMachine/contracts/VendingMachine.ral [hunk 5]
```diff
@@ -83,42 +79,37 @@ Contract VendingMachine(
...
-
-        emit NftMinted(minter, contractId, 1)
-        return toByteVec!(foodType) // return contract id later
+        let minter = callerAddress!()
+        let mut contractIds = #
+        let mut startingIndex = 0
+        let mut nftIdToMint = 0
+
+        for (let mut i = 0; i < mintAmount; i = i + 1) {
+            nftIdToMint = pickIdForFood(foodTypeId_)
```
Comment (not-reed) [inline apps/VendingMachine/contracts/VendingMachine.ral:99]: does this work if `mintAmount` is 10, but there are only 5 supply available? what happens?

Comment (rabTAI) [reply apps/VendingMachine/contracts/VendingMachine.ral:99]: ` pub fn mint(foodTypeId_: U256,mintAmount: U256) -> ByteVec {
        assert!(!mintIsPaused, ErrorCodes.MintIsPaused)
        assert!(totalSupply + mintAmount <= MaxSupply, ErrorCodes.MaxSupplyReached)
        assert!(mintAmount <= MaxMintPerTx, ErrorCodes.MaxPerTx)`
` assert!(totalSupply + mintAmount <= MaxSupply, ErrorCodes.MaxSupplyReached)`

File: apps/VendingMachine/contracts/VendingMachine.ral (hunk 2 above)
Comment (rabTAI) [reply apps/VendingMachine/contracts/VendingMachine.ral:15]: Good catch, I will make it a mutable. I thought it was a collection name

File: apps/VendingMachine/tsconfig.json (hunk 3 above)
Comment (rabTAI) [reply apps/VendingMachine/tsconfig.json:24]: Removed

File: apps/VendingMachine/test/VendingMachine.test.ts (hunk 4 above)
Comment (rabTAI) [reply apps/VendingMachine/test/VendingMachine.test.ts:172]: passed Id and worked

**Pull request merged**